# Ejemplo 3.5.3 Diaz & Morales (2015)

<img src="https://raw.githubusercontent.com/fhernanb/fhernanb.github.io/master/my_docs/logo_unal_color.png" alt="drawing" width="200"/>

En este ejemplo se desea estudiar el siguiente conjunto de hipótesis usando un nivel de significancia del 0.05.

$H_0: \boldsymbol{\delta} = (0, 0)^\top$

$H_A: \boldsymbol{\delta} \neq (0, 0)^\top$

usando los siguientes datos

<img src="datos_tuberias_pareados.png" alt="" width="800"/>

In [1]:
import numpy as np
import scipy.stats

A continuación los datos del ejemplo

In [2]:
de1 = np.array((22, 2, 4, 12, 11, 15, 28, -5, 8, 4, -1, -10, -2, 25, 7))
de2 = np.array((-4, 5, 3, -3, 2, 4, 10, -1, 10, 6, 5, 0, -4, 8, 5))

dt = np.column_stack((de1, de2))
print(dt)

[[ 22  -4]
 [  2   5]
 [  4   3]
 [ 12  -3]
 [ 11   2]
 [ 15   4]
 [ 28  10]
 [ -5  -1]
 [  8  10]
 [  4   6]
 [ -1   5]
 [-10   0]
 [ -2  -4]
 [ 25   8]
 [  7   5]]


<div class="alert alert-danger">
  <strong>Cuidado!</strong> Se debe verificar que la muestra aleatoria de diferencias $d_{e1}$ y $d_{e2}$ provenga de una población normal multivariada. Aquí vamos a asumir que este supuesto se cumple.
</div>

El vector de medias se obtiene así:

In [3]:
d_barra = np.mean(dt, axis=0)   # Para obtener el vector de medias
d_barra = np.matrix(d_barra).T   # Para convertir a matriz columna
print(d_barra)                   # Para ver el contenido
print(d_barra.shape)             # Para ver la dimensión (shape)                           # Para ver el contenido

[[8.        ]
 [3.06666667]]
(2, 1)


La matriz de varianzas se obtiene así:

In [4]:
Sd = np.cov(dt.T) # Para obtener la matriz
print(Sd)          # Para ver el contenido
print(Sd.shape)    # Para ver la dimensión (shape)

[[121.57142857  17.07142857]
 [ 17.07142857  21.78095238]]
(2, 2)


Ahora calculamos $T^2=n (\bar{\mathbf{D}}-\boldsymbol{\delta}_0)^\top \boldsymbol{S}_d^{-1} (\bar{\mathbf{D}}-\boldsymbol{\delta}_0)$.

In [5]:
n = 15 # número de observaciones
p = 2  # dimensión (variables diferencia)

Sd_inv = np.linalg.inv(Sd)
T2 = n * np.matmul(np.matmul(d_barra.transpose(), Sd_inv), d_barra)
T2.item()  # Para obtener el T2 como un escalar y no como una matriz

10.81889854019835

El valor crítico para realizar la prueba es $\frac{(n-1)p}{(n-p)} F_{p, n-p}(\alpha)$. Para obtener el valor crítico con $\alpha=0.05$ se procede así:

In [6]:
p * (n-1) * scipy.stats.f.ppf(q=1-0.05, dfn=p, dfd=n-p) / (n-p)   # Valor crítico

8.196602083337353

<div class="alert alert-success">
  <strong>Resultado</strong> Como el estadístico es $T^2=10.82$, que es mayor que el valor crítico $8.20$, entonces hay evidencias suficientes para rechazar $H_0: \boldsymbol{\delta} = (0, 0)^\top$.
</div>